include/unifex/async_scope.hpp

```diff
@@ -655,53 +653,100 @@ class _attached_op<Sender, Receiver>::type final {
   ~type() {
     auto scope = scope_ref();
     if (scope) {
-      unifex::deactivate_union_member(op_);
+      op_.destruct();
       // started => receiver responsible for cleanup
-      if (!has_started()) {
+      if (incref() == 1) {
         record_done(scope);
       }
-    } else {
-      unifex::deactivate_union_member(receiver_);
     }
   }
 
-  auto get_token() { return stopSource_.get_token(); }
+  void request_stop() noexcept {
+    // increment from 0 means lost race with a completion method so
+    // no-op and let the in-flight completion complete
+    if (incref() == 0) {
+      return;
+    }
```

Comment:
We discovered while pairing that this `incref()` is a bug; we need a `compare_exchange_strong` that leaves the reference count untouched if it's already zero.